Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py

```diff
@@ -19,16 +22,27 @@
 class WebServerBenchmarkRunner(BenchmarkRunner):
     name = 'webserver'
 
-    def __init__(self, plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit=True, show_iteration_values=False, device_id=None, diagnose_dir=None):
+    def __init__(self, plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit=True, show_iteration_values=False, device_id=None, diagnose_dir=None, profile_output_directory=None):
         self._http_server_driver = HTTPServerDriverFactory.create(platform)
         self._http_server_driver.set_device_id(device_id)
-        super(WebServerBenchmarkRunner, self).__init__(plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit, show_iteration_values, device_id, diagnose_dir)
+        self._profile_output_directory = profile_output_directory
+        super(WebServerBenchmarkRunner, self).__init__(plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit, show_iteration_values, device_id, diagnose_dir, profile_output_directory)
         if self._diagnose_dir:
             self._http_server_driver.set_http_log(os.path.join(self._diagnose_dir, 'run-benchmark-http.log'))
 
     def _get_result(self, test_url):
         result = self._browser_driver.add_additional_results(test_url, self._http_server_driver.fetch_result())
         assert(not self._http_server_driver.get_return_code())
+
+        if self._profile_output_directory:
+            print(f"Getting benchmark profile results for {test_url} and copying them to {self._profile_output_directory}.")
+            copy_output = subprocess.Popen(r"""log stream --style json --color none | perl -mFile::Basename -mFile::Copy -nle 'if (m/<WEBKIT_LLVM_PROFILE>.*<BEGIN>(.*)<END>/) { (my $l = $1) =~ s/\\\//\//g; my $b = File::Basename::basename($l); my $d = """ + "\"" + self._profile_output_directory + """/$b"; print "Moving $l to $d"; File::Copy::move($l, $d); }'""", shell=True, bufsize=0)
+            time.sleep(1)
+            os.system("notifyutil -p com.apple.WebKit.profiledata")
+            time.sleep(7)
+            copy_output.kill()
+            time.sleep(1)
+            print("Hopefully the benchmark profile has finished writing to disk, moving on.")
```

Comment:
I originally tried this, but had too much trouble and gave up. Reading from stdout blocks so badly that even a timer couldn't fix it. This does obviously seem like the better approach, but I couldn't find a way around readline blocking. If you have any suggestions, it may be worth a shot

Reply:
Ugh, yeah. Maybe passing `('--pid', self._browser_driver._safari_process.pid)` would cut down the log noise enough that Python can handle it. But the real answer is that Python is not great at these kinds of tasks. No worries if the shell pipeline ends up working better; at some point we're jumping through too many hoops in python to make it worth refactoring.